test/e2e/dap_minimal_test.go

```diff
@@ -0,0 +1,152 @@
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build e2e
+// +build e2e
+
+package e2e
+
+import (
+	"context"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/e2e"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/environments"
+	apicommon "github.com/DataDog/datadog-operator/api/datadoghq/common"
+	"github.com/DataDog/datadog-operator/test/e2e/common"
+	"github.com/DataDog/datadog-operator/test/e2e/provisioners"
+	"github.com/DataDog/datadog-operator/test/e2e/tests/utils"
+
+	"github.com/DataDog/test-infra-definitions/components/datadog/agentwithoperatorparams"
+	"github.com/DataDog/test-infra-definitions/components/datadog/operatorparams"
+	"github.com/stretchr/testify/assert"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+var (
+	dapNameRegex  = "^datadog-agent-with-profile-[A-Za-z0-9-]+"
+	dapHelmValues = `installCRDs: false
+datadogAgentProfile:
+  enabled: true`
+)
+
+type localKindDAPSuite struct {
+	e2e.BaseSuite[environments.Kubernetes]
+}
+
+func TestLocalKindDAPSuite(t *testing.T) {
+	operatorOptions := []operatorparams.Option{
+		operatorparams.WithNamespace(common.NamespaceName),
+		operatorparams.WithOperatorFullImagePath(common.OperatorImageName),
+		operatorparams.WithHelmValues(dapHelmValues),
+	}
+
+	ddaConfigPath, err := common.GetAbsPath(filepath.Join(manifestsPath, "datadog-agent-resources.yaml"))
+	assert.NoError(t, err)
+
+	ddaOptions := []agentwithoperatorparams.Option{
+		agentwithoperatorparams.WithNamespace(common.NamespaceName),
+		agentwithoperatorparams.WithDDAConfig(agentwithoperatorparams.DDAConfig{
+			Name:         "datadog-resources",
+			YamlFilePath: ddaConfigPath,
+		}),
+	}
+
+	provisionerOptions := []provisioners.KubernetesProvisionerOption{
+		provisioners.WithK8sVersion(common.K8sVersion),
+		provisioners.WithOperatorOptions(operatorOptions...),
+		provisioners.WithDDAOptions(ddaOptions...),
+		provisioners.WithYAMLWorkload(provisioners.YAMLWorkload{Name: "dap", Path: filepath.Join(manifestsPath, "dap.yaml")}),
+		provisioners.WithoutFakeIntake(),
+	}
+
+	e2eOpts := []e2e.SuiteOption{
+		e2e.WithProvisioner(provisioners.KubernetesProvisioner(provisionerOptions...)),
+	}
+
+	e2e.Run(t, &localKindDAPSuite{}, e2eOpts...)
+}
+
+func (s *localKindDAPSuite) TestDAP() {
+	s.T().Run("Verify Operator", func(t *testing.T) {
+		s.Assert().EventuallyWithT(func(c *assert.CollectT) {
+			utils.VerifyOperator(s.T(), c, common.NamespaceName, s.Env().KubernetesCluster.Client())
+		}, 300*time.Second, 15*time.Second, "Could not validate operator pod in time")
+	})
+
+	s.T().Run("Apply DAP to single node", func(t *testing.T) {
```

Comment:
Could you move this run command and the associated helper functions to `kind_test.go`? That way it will run in ci with all the other e2e tests. I think the dap file can be added using `k8s.KubectlApply` and you can verify with the DDA label that it's the right config to run the dap test